kuksa_databroker/databroker/src/broker.rs

```diff
@@ -175,14 +177,29 @@ impl Entry {
     pub fn validate(&self, update: &EntryUpdate) -> Result<(), UpdateError> {
         if let Some(datapoint) = &update.datapoint {
             self.validate_value(&datapoint.value)?;
+            self.validate_allowed(&datapoint.value)?;
         }
         if let Some(Some(value)) = &update.actuator_target {
             self.validate_value(value)?;
+            self.validate_allowed(value)?;
         }
         Ok(())
     }
 
-    fn validate_value(&self, value: &DataValue) -> Result<(), UpdateError> {
+    fn validate_allowed(&self, value: &DataValue) -> Result<(), UpdateError> {
+        // check if allowed value
+        if let Some(allowed_values) =  &self.metadata.allowed {
+            match allowed_values.find(value){
```

Comment:
If the `value` is of array type (i.e. any of the `DataValue::*Array`) the correct behaviour is to check whether each element of the array is present in `allowed`. See https://covesa.github.io/vehicle_signal_specification/rule_set/data_entry/allowed/#allowed-values-for-array-types.

Hence the validation check needs to account for this.